04/app.js

```diff
@@ -10,14 +11,31 @@ class App extends React.Component {
     }
 
     renderUsersList() {
-        const {users} = this.state;
-        return users.map(name => {
-            return (
+        const { users, searchQuery } = this.state;
+        if(searchQuery === '') {
+            return this.mapUserList(users, 'map')
+        }
+        else {
+            return this.mapUserList(users, 'filter', searchQuery);
+        }
+    }
+
+    mapUserList = (arr, myType, query) => {
+        if(myType === 'map') {
```

Comment:
Osobiście napisałbym osobne funkcje. Jedna miałaby `map`, druga `filter` i wykorzystał w odpowiednim miejscu.
To rozwiązanie wprowadza do kodu niepotrzebne if-y ;)